uamqp/link.py

```diff
@@ -108,6 +114,23 @@ def from_incoming_frame(cls, session, handle, frame):
         # check link_create_from_endpoint in C lib
         raise NotImplementedError('Pending')  # TODO: Assuming we establish all links for now...
 
+    def get_state(self):
+        if self.state.value == 6:  # error
+            raise self._error
+        return self.state
+
+    def _process_link_error(self, condition, description, info):
+        try:
+            amqp_condition = ErrorCodes(condition)
+        except ValueError:
+            error = VendorLinkDetach(condition, description, info)
+        else:
+            if amqp_condition == ErrorCodes.LinkRedirect:
+                error = LinkRedirect(amqp_condition, description, info)
+            else:
+                error = LinkDetach(amqp_condition, description, info)
```

Comment:
Same comment as connection - I think this function can be replaced with a simple:
```python
self._error = LinkRedirect(...) if condition == LinkRedirect else LinkDetach(...)
```

Reply:
code updated to be like:
```
            error_cls = LinkRedirect if frame[2][0] == ErrorCodes.LinkRedirect else LinkDetach
            self._error = error_cls(condition=frame[2][0], description=frame[2][1], info=frame[2][2])
```